src/Analyser/TypeSpecifier.php

```diff
@@ -2149,10 +2157,19 @@ public function resolveIdentical(Expr\BinaryOp\Identical $expr, Scope $scope, Ty
 			}
 		}
 
-		if (count($rightType->getConstantStrings()) > 0) {
+		if ($rightType->isInteger()->yes() || $rightType->isString()->yes()) {
 			$types = null;
-			foreach ($rightType->getConstantStrings() as $constantString) {
-				$specifiedType = $this->specifyTypesForConstantStringBinaryExpression($unwrappedLeftExpr, $constantString, $context, $scope, $rootExpr);
+			foreach ($rightType->getFiniteTypes() as $finiteType) {
```

Comment:
I need to check whether this leads to dead code in IntegerRange handling somewhere